src/routes/Profile/Profile.tsx

```diff
@@ -55,30 +50,73 @@ const Profile: React.FunctionComponent = () => {
     }
     setCollectionBadges(result);
     setClaimButtonBulkMode('claim');
-    setClaimButtonModes(
-      SEASON_BADGE_VARIANTS[season.id].reduce(
-        (pV, cI) => ({
-          ...pV,
-          [cI as BadgeVariant]: 'claim',
-        }),
-        {} as Record<BadgeVariant, ClaimButtonProps['mode']>,
+    const claimedTodayVariants = SEASON_BADGE_VARIANTS[season.id].filter((badgeVariant) =>
```

Comment:
why do we need to compute both claimedTodayVariants and claimedInThePastVariants ?

Reply:
due to business logic. badges claimed today should only show checkbox + claimed text. badges claimed in the past should show UI with Claimed: <DATE>. It is just for the mode